packages/search/src/content/index.ts

```diff
@@ -89,13 +89,195 @@ export class ContentSearchService
  * A standalone function for searching content across the Portal API only or Portal API and the
  * Hub Indexer V3 API.
  *
- * @param request - the IContentSearchRequest instance for searching
+ * The `filter` argument allows the caller to filter results based on attributes exposed by AGO and
+ * the Hub API. A complete list of attributes can be found at the docs for `IContentSearchFilter`,
+ * but some examples include:
+ * - access
+ * - id
+ * - group
+ * - orgid
+ *
+ * The `options` argument allows the caller to specify more general attributes about the request itself.
+ * A complete list of options can be found at the docs for `IContentSearchFilter`, but some examples include:
+ * - the authenticated user
+ * - whether the call is happening in an enterprise environment
+ * - what site catalog to search
+ *
+ * Combined, both filters and options allow you to create complex queries against AGO / Hub API.
+ * Here are examples of some common use cases:
  *
  * ```js
- * import { searchContent } from '@esri/hub-search'
+ * import { searchContent, IBooleanOperator } from '@esri/hub-search';
+ * ...
+ *
+ * /////////////////////////////////////////////////////////////////////
+ * // Search for all public and private content associated with a site
+ * /////////////////////////////////////////////////////////////////////
+ * const options: IContentSearchOptions = {
+ *    site: 'https://my-site.hub.arcgis.com',
+ *    portal: 'https://www.arcgis.com',
+ *    // Note: only private content visible to the authenticated user will be returned
```

Comment:
This note could be misinterpreted to mean that no public content will be returned.
```suggestion
 *    // Note: no private content that is unavailable to the authenticated user will be included
```

Reply:
Good catch! I didn't even think about that!

Reply:
Hate to be that guy, but including both `no` and `unavailable` seems like a double negative. How about `All public, as well as private content accessible to authenticated user, will be included`?

Reply:
Fair point. Not to be "that other guy" but what about

`Any private content that the authenticated user can access will be included in the results`

Reply:
Personally, I like that suggestion @andrewctate because it will flow a little better as an inline comment and it's scoped to just the private content